src/VolumeLogger.js

```diff
@@ -59,6 +59,295 @@ module.exports = class VolumeLogger {
+                    } else {
+                        minReport.broker.messagesToNetworkPerSec = throtheledAvg(minReport.broker.messagesToNetworkPerSec, secReport.broker.messagesToNetworkPerSec)
+                        minReport.broker.bytesToNetworkPerSec = throtheledAvg(minReport.broker.bytesToNetworkPerSec, secReport.broker.bytesToNetworkPerSec)
+
+                        minReport.network.avgLatencyMs = throtheledAvg(minReport.network.avgLatencyMs, secReport.network.avgLatencyMs)
+                        minReport.network.bytesToPeersPerSec = throtheledAvg(minReport.network.bytesToPeersPerSec, secReport.network.bytesToPeersPerSec)
+                        minReport.network.bytesFromPeersPerSec = throtheledAvg(minReport.network.bytesFromPeersPerSec, secReport.network.bytesFromPeersPerSec)
+                        minReport.network.connections = throtheledAvg(minReport.network.connections, secReport.network.connections)
+
+                        if (metricsReport.metrics['broker/cassandra']) {
+                            minReport.storage.bytesWrittenPerSec = throtheledAvg(minReport.storage.bytesWrittenPerSec, secReport.storage.bytesWrittenPerSec)
+                            minReport.storage.bytesReadPerSec = throtheledAvg(minReport.storage.bytesReadPerSec, secReport.storage.bytesReadPerSec)
+                        }
+                    }
+
+                    securePublish(
+                        this.streamIds.secStreamId,
+                        secReport
+                    )
+
+                    const now = Date.now()
+
+                    if (sec === 60 || (minReport.timestamp + (60 * 1000) - now) < 0) {
+                        minReport.timestamp = secReport.currentTime
+                        sec = 0
+
+                        securePublish(
+                            this.streamIds.minStreamId,
+                            minReport
+                        )
+
+                        // let's check, once a minute, if it's time to run hour and/or daily reports
+
+                        // get the last msg to check if it's been an hour
+
+                        const lastHourReports = await getResend(this.streamIds.hourStreamId, 1)
+
+                        if (lastHourReports.length === 0) {
+                            securePublish(
+                                this.streamIds.hourStreamId,
+                                hourReport
+                            )
+                        } else if ((lastHourReports[0].timestamp + (60 * 60 * 1000) - now) < 0) {
+                            // fetch the last 60 minute reports and get the averages
+                            const messages = await getResend(this.streamIds.minuteStreamId, 60)
+
+                            for (let i = 1; i < messages.length; i++) {
+                                hourReport.broker.messagesToNetworkPerSec = throtheledAvg(hourReport.broker.messagesToNetworkPerSec, messages[i].broker.messagesToNetworkPerSec)
```

Comment:
We may want to calculate the average here using a full-window approach. Let's discuss.